tests/src/Context/DrupalContext.php

```diff
@@ -156,4 +152,92 @@ public function assertUserExistence($username) {
+   *   The membership table.
+   *
+   *   Table format:
+   *   | group_type   | group_id  | member    |
+   *   | entityTypeId | entity_id | user name |
+   *   | entityTypeId | entity_id | user name |
+   *
+   *   Only the name field is required.
+   *
+   * @throws \Exception
+   *   Thrown when an entity is not found or when an entity set as group is not
+   *   a group.
+   *
+   * @Given (the following )user memberships:
+   */
+  public function givenUserMembership(TableNode $membership_table) {
+    foreach ($membership_table->getColumnsHash() as $membership_row) {
+      foreach ($membership_row as $key => $value) {
+        $values[$key] = $value;
+      }
+
+      // Load group.
+      $group = \Drupal::entityTypeManager()
+        ->getStorage($values['group_type'])
+        ->load($values['group_id']);
+      if (empty($group)) {
+        throw new \Exception("Entity " . $values['group_id'] . " not found.");
+      }
+
+      // Load member.
+      $member = user_load_by_name($values['member']);
+      if (empty($member)) {
+        throw new \Exception("Entity " . $values['group_id'] . " not found.");
+      }
+
+      /** @var \Drupal\og\OgMembership $membership */
+      $membership = Og::membershipStorage()->create(Og::membershipDefault());
+      $membership
+        ->setEntityType($group->getEntityTypeId())
+        ->setEntityid($group->id())
+        ->setState(OgMembershipInterface::STATE_ACTIVE);
+      if ($member->getEntityTypeId() == 'user') {
+        $membership->setUser($member->id());
+      }
+      $membership->save();
+    }
+  }
+
+  /**
+   * Checks if the given entity is member of the given group.
+   *
+   * @param string $member_title
+   *   The member entity's label.
+   * @param string $member_type
+   *   The member entity's bundle.
+   * @param string $group_id
+   *   The group entity's label.
+   *
+   * @throws \Exception
+   *   Thrown when an entity is not found or when an entity set as group is not
+   *   a group.
+   *
+   * @Then the :member_label of type :member_type should be a member of the group with ID :group_id
+   */
+  public function assertOgMembership($member_title, $member_type, $group_id) {
+    $member_storage = \Drupal::entityTypeManager()->getStorage($member_type);
+    $members_array = $member_storage->loadByProperties(['title' => $member_title]);
+    /** @var \Drupal\core\Entity\EntityInterface $member */
+    $member = reset($members_array);
+
+    if (empty($member)) {
+      throw new \Exception("The entity titled $member_title was not found.");
+    }
+    if (!(Og::isGroupContent($member->getEntityTypeId(), $member->bundle()))) {
+      throw new \Exception("The entity " . $member->getEntityTypeId() . " is not a group content.");
+    }
+
+    if ($member->og_group_ref->getValue()[0]['target_id'] != $group_id) {
+      throw new \Exception("Entity \"" . $member->label() . "\" is not a member of entity with id \"" . $group_id . "\"");
+    }
```

Comment:
This assumes that the field name of the groups audience field is always called `og_group_ref` but this is not always the case, in can have any name. You also need to special case users, since they do not use field storage.

Anyway it's not necessary to do all this stuff, you can use `Og::isMember()` to do the check. But from a quick look it seems this is not yet updated to discern between users and other entities either :(